### ISSUE TITLE
Room booking API: Support for recurring bookings / pre-bookings

### ISSUE THREAD

**Issue by olifre** [indico/indico#5787]
**Is your feature request related to a problem? Please describe.**
The room booking API currently has two major limitations:
* No recurrent bookings possible. 
* No pre-bookings possible (i.e. only "actual" bookings / API user needs booking privilege). 

Our use case is that a different authoritative room booking system for teaching activities is used at our site (as is the case with most Universities). Such booking systems are usually not well-made for events, so for all non-teaching related bookings (and rooms not used for teaching activities), we want to use Indico, but teaching bookings take priority. 
Via the booking API, we want to carry over all "teaching bookings" automatedly. 

**Describe the solution you'd like**
Essentially, making the room booking API full-featured ;-). 

**Describe alternatives you've considered**
De-serializing recurring bookings (which makes it harder to modify them later on, e.g. change the title), or booking manually (which is more effort...).

**Comment by ThiefMaster** [comment]
Yeah, the current room booking API is pretty crappy. I recommend using the internal APIs which are also used by the React front-end. Simply create a personal access token with the "everything (all methods)" scope and you can use that to make authenticated calls to those internal APIs.

**Comment by olifre** [comment]
Thanks, I think we will do that for now indeed :+1: . Of course, in the long-term, it would be nice if the official API would offer this, too.

**Comment by ThiefMaster** [comment]
TBH, I think if we make a more official room booking API at some point there's a good chance that we'd reuse the internal ones with none or just minor changes. I think the main thing to introduce would be a "roombooking" scope for tokens.

**Comment by olifre** [comment]
This would indeed be ideal, so even if we use the internal APIs now, there would be no huge changes later on :+1: .
